### PR TITLE
Add names and originalFileNames to assets

### PR DESCRIPTION
**Pull request by lukastaegert**
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### Description
When no explicit "fileName" is provided when emitting an asset, Rollup will associate assets with the same content. At the moment, however, this is not reflected in the bundle and the `output.assetFileNames` option. While every file emission can have a different `name` and `originalFileName` property, only one of them is visible via the `.name` and `.originalFileName` properties, see https://github.com/rollup/rollup/pull/5596#issuecomment-2282746429
This PR adds new `.names` and `.originalFileNames` properties that are arrays and contain all emitted names. At the same time, the old properties are soft-deprecated, meaning that accessing them will cause a warning in the next major version of Rollup.

cc @sapphi-red

### CODE REVIEW

File: src/rollup/types.d.ts [hunk 1]
```diff
@@ -815,8 +815,12 @@ export type WarningHandlerWithDefault = (
 export type SerializedTimings = Record<string, [number, number, number]>;
 
 export interface PreRenderedAsset {
+	/** @deprecated Use "originalFileNames" instead. */
```
Comment (sapphi-red) [inline src/rollup/types.d.ts:818]:
```suggestion
	/** @deprecated Use "names" instead. */
```

**Comment by sapphi-red** [comment]
Thanks! I tried this PR at https://github.com/vitejs/vite/pull/18240. It works for Vite's case 👍

File: src/rollup/types.d.ts (hunk 1 above)
Comment (lukastaegert) [reply src/rollup/types.d.ts]: Thanks for spotting!

**Comment by prisis** [comment]
Hey @lukastaegert, how does the migration looks from name to names, before is was using 

example:
```
    assetFileNames({ name }) {
        const p = "[name][extname]";

        if (!name) {
            return p;
        }

        if (name.endsWith(".css")) {
            return `css/${p}`;
        }

        if (name.endsWith(".map")) {
            return `map/${p}`;
        }

        return p;
    },
```

with names do i need only to check for the first entry, or do i need to check all entries in names?

**Comment by lukastaegert** [comment]
Technically you would need to check all entries, though it will be one entry most of the time anyway. I usually would not expect the extensions to differ if the content is the same, and there is probably no reason to pick any but the first name if your just need a name and do not want to compare it to anything.

**Pull request merged**